generator/osm2meta.cpp

```diff
@@ -5,7 +5,136 @@
+{
+  if (!ftypes::IsSpeedCamChecker::Instance()(m_params.m_Types))
+    return string();
+
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_stars(string const & v) const
+{
+  if (v.empty())
+    return string();
+
+  // We are accepting stars from 1 to 7.
+  if (v[0] <= '0' || v[0] > '7')
+    return string();
+
+  // Ignore numbers large then 9.
+  if (v.size() > 1 && ::isdigit(v[1]))
+    return string();
+
+  return string(1, v[0]);
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_operator(string const & v) const
+{
+  auto const & isATM = ftypes::IsATMChecker::Instance();
+  auto const & isFuelStation = ftypes::IsFuelStationChecker::Instance();
+
+  if (!(isATM(m_params.m_Types) || isFuelStation(m_params.m_Types)))
+    return string();
+
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_url(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_phone(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_opening_hours(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_ele(string const & v) const
+{
+  auto const & isPeak = ftypes::IsPeakChecker::Instance();
+  if (!isPeak(m_params.m_Types))
+    return string();
+  double val = 0;
+  if(!strings::to_double(v, val) || val == 0)
+    return string();
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_turn_lanes(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_turn_lanes_forward(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_turn_lanes_backward(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_email(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_postcode(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_flats(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_height(string const & v) const
+{
+  double val = 0;
+  string corrected(v, 0, v.find(" "));
+  if(!strings::to_double(corrected, val) || val == 0)
+    return string();
+  ostringstream ss;
+  ss << fixed << setprecision(1) << val;
+  return ss.str();
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_building_levels(string const & v) const
+{
+  double constexpr kMetersPerLevel = 3;
+  double val = 0;
+  if(!strings::to_double(v, val) || val == 0)
+    return string();
+  ostringstream ss;
+  ss << fixed << setprecision(1) << (val * kMetersPerLevel);
+  return ss.str();
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_denomination(string const & v) const
+{
+  return v;
+}
+
+string MetadataTagProcessorImpl::ValidateAndFormat_cuisine(string v) const
+{
+  strings::MakeLowerCaseInplace(v);
+  v = regex_replace(v, regex("(;|,)( *)"), ";");
```

Comment:
You do not need second group, right? I mean, the regexp may be simplified like this:
```c++
[;,] *
```